Week 1 683380574-6 นางสาวจันทรณภา จอมทอง

Lab 1.1: Environment Setup & First Script

In [ ]:
print("Hello,Word!")

Hello,Word!


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [ ]:
age = 21
gpa = 3.64
favoritecolor = "Denim blue"
Student = True
major = "cp-cs"

print("Age: ",age,type(age))
print("My GPA: ",gpa,type(gpa))
print("My favorite color: ",favoritecolor,type(favoritecolor))
print("Student: ",Student,type(Student))
print("My major: ",major,type(major))

Age:  21 <class 'int'>
My GPA:  3.64 <class 'float'>
My favorite color:  Denim blue <class 'str'>
Student:  True <class 'bool'>
My major:  cp-cs <class 'str'>


3. Perform simple arithmetic:


In [ ]:
Num1 = 67
Num2 = 68
result1 = Num1+Num2
result2 = Num1-Num2
result3 = Num1*Num2
result4 = Num1/Num2



print("Result of Num1+Num2 is: ",result1)
print("Result of Num1-Num2 is: ",result2)
print("Result of Num1-Num2 is: ",result3)
print("Result of Num1-Num2 is: ",result4)

Result of Num1+Num2 is:  135
Result of Num1-Num2 is:  -1
Result of Num1-Num2 is:  4556
Result of Num1-Num2 is:  0.9852941176470589


4. String concatenation and f-strings:


In [ ]:
firtname = "jantaranapha"
lastname = "Jomthong"

Fullname = firtname+" "+lastname
print("Hello!",Fullname)

Hello! jantaranapha Jomthong


5. User input and type conversion:


In [ ]:
Num = input("What your favorite Number: ")
print("check type1: ",type(Num))
Num = int(Num)
result = Num+5
print("Final result is: ",result)
print("Check type2: ",type(Num))

What your favorite Number: 12
check type1:  <class 'str'>
Final result is:  17
Check type2:  <class 'int'>


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: การนำไปประยุกต์ใช้ในชีวิตประจำวันนั้นมีหลากหลายมาก เช่น การจัดไฟล์อัตโนมัติ, การกรอกฟอร์มอัตนมัติ, การคำนวณภาษี เเละอีกมากมาย

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: 13.5 type float

ทำไมถึงเป็นtype float เพราะในpythonเมื่อนำ int+floatจะเป็น float

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์:Alice is 30 years old. type string

ทำไมถึงเป็นtype string เพราะว่าเราประกาศตัวเเปรให้เป็นStringดังนั้นการบวกกันก็เหมือนการต่อข้อความ

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์:False type boolean

ทำไมถึงเป็นtype boolean เพราะว่าเรากำหนดtypeให้is_sunnyเป็นboolean

In [ ]:
# Puzzler 4
# What happens here?
num_str = input("Enter a number: ")
result = num_str + 5
print(result)

KeyboardInterrupt: Interrupted by user

ได้ผลลัพธ์: error

เพราะว่าค่า num_str เป็น strดังนั้นนำไปบวกกับint เลยerror